app/actions/handlers.py

```diff
@@ -18,24 +18,73 @@
 RUMI_BASE_URL = "https://innogando-backend-prod-01.innogando.com"
 
 
-def transform(farm, observation):
+def transform(farm, animals_info, observation):
+    rumi_id_map = {
+        animal["rumi_id"]: {"type": animal_type, **animal}
+        for animal_type, animal_type_info in animals_info.items()
+        for animal in animal_type_info
+    }
+
+    animal_info = rumi_id_map.get(observation.device_name, None)
+
+    source_name = (
+        f"{animal_info.get('name', observation.official_tag)} ({animal_info.get('rumi_id')})"
+       if animal_info
+       else
+       f"{observation.official_tag} ({observation.device_name})"
+    )
+
+    subject_type = f"rumi-{animal_info.get('type')}" if animal_info else "tracking-device"
+
+    additional_info = {
+        key: value for key, value in (animal_info or {}).items() if value
+    }
+
     return {
-        "source_name": observation.device_name,
+        "source_name": source_name,
         "source": observation.official_tag,
-        "type": "tracking-device",
-        "subject_type": "vehicle",
+        "type": subject_type,
+        "subject_type": "unassigned",
```

Comment:
Let's leave `type` equal to "tracking-device".
And use subject_type to hold the cow, bull or calf designation.

```suggestion
        "type": "tracking-device",
        "subject_type":   "rumi-cow" | "rumi-bull" | "rumi-calf"
```